tests/conftest.py

```diff
@@ -55,4 +69,82 @@ def tls_certificate_pem_bytes(tls_certificate):
 @pytest.fixture
 def tls_certificate_fingerprint_sha256(tls_certificate_pem_bytes):
     tls_cert_der = ssl.PEM_cert_to_DER_cert(tls_certificate_pem_bytes.decode())
-    return hashlib.sha256(tls_cert_der).digest()
+    return sha256(tls_cert_der).digest()
+
+
+@pytest.fixture
+@needs_unix
+def unix_sockname(tmp_path, tmp_path_factory):
+    """Generate an fs path to the UNIX domain socket for testing.
+
+    N.B. Different OS kernels have different fs path length limitations
+    for it. For Linux, it's 108, for HP-UX it's 92 (or higher) depending
+    on its version. For for most of the BSDs (Open, Free, macOS) it's
+    mostly 104 but sometimes it can be down to 100.
+
+    Ref: https://github.com/aio-libs/aiohttp/issues/3572
+    """
+    max_sock_len = 92 if IS_HPUX else 108 if IS_LINUX else 100
+    """Amount of bytes allocated for the UNIX socket path by OS kernel.
+
+    Ref: https://unix.stackexchange.com/a/367012/27133
+    """
+
+    sock_file_name = 'unix.sock'
+
+    root_tmp_dir = Path('/tmp').resolve()
+    os_tmp_dir = Path(os.getenv('TMPDIR', '/tmp')).resolve()
+    original_base_tmp_path = Path(tmp_path_factory.getbasetemp())
```

Comment:
Is `Path()` convertion required?

Reply:
It's just nice for the unified API and paths comparison.